### PR TITLE
Add Event Emitter

### PR DESCRIPTION
**Pull request by daniel-dressler**

### CODE REVIEW

File: EventEmitter/EventEmitter.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+			return;
+		}
+
+		Action<object, T> execEventList = (Action<object, T>)eventsList[eventTags[eventTag]];
+		execEventList(sender, arguments);
+	}
+	
+	public void emit(string eventTag, T arguments)
+	{
+		emit (eventTag, null, arguments);
```
Comment (ronkorving) [inline EventEmitter/EventEmitter.cs]: overlooked a whitespace there

**Comment by nullorvoid** [comment]
Except styles this looks fine to me, although I still don't like the idea of a global event emitter.

**Comment by ronkorving** [comment]
I think EventEmitter without a `once` is crippled. People need a `once` function in their emitter or will most definitely hack around it. Not desirable.

File: EventEmitter/EventEmitter.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+	//
+	public void emit(string eventTag, object sender, T arguments)
+	{
+		if (!eventTags.ContainsKey(eventTag)) {
+			return;
+		}
+
+		Action<object, T> execEventList = (Action<object, T>)eventsList[eventTags[eventTag]];
+		execEventList(sender, arguments);
```
Comment (ronkorving) [inline EventEmitter/EventEmitter.cs]: What is execEventList?

**Comment by AlmirKadric** [comment]
@nullorvoid yeah there is a conceptual shift using this
after playing with C#, having predefined event types inside the class and invoking or listening to them is nice. Just means we have to predefine all of our events rather than using a single emitter. We can try a refactor later and see if it nicer to play with. For now this is a requirement till then

**Comment by AlmirKadric** [comment]
@ronkorving added the missing once function

**Comment by ronkorving** [comment]
Awesome, thanks :)

**Comment by ronkorving** [comment]
I don't wanna hold this up too much, and I'm fine if issues are made for it. But I think EventEmitter would also deserve `listenerCount(eventName)`, `off(eventName, hander)` (aka: removeListener) and `removeAllListeners(optional eventName)`.

**Pull request merged**